ContextDesktop/src/main/java/com/github/migi_1/Context/model/MainEnvironment.java

```diff
@@ -176,15 +190,16 @@ private void initSpatials() {
         commander = new Commander(COMMANDER_LOCATION);
         carriers = createCarriers();
         damageDealerGenerator = new DamageDealerGenerator(commander);
-
         //attach all objects to the root pane
         for (LevelPiece levelPiece : levelGenerator.getLevelPieces(COMMANDER_LOCATION)) {
             addDisplayable(levelPiece);
         }
 
         for (DamageDealer damageDealer : damageDealerGenerator.getObstacles()) {
-            damageDealer.collideWith(platform.getModel().getWorldBound(), results);
-            addDisplayable(damageDealer);
+            for (Entry<Entity, CollisionResults> entry : results.entrySet()) {
+                damageDealer.collideWith(entry.getKey().getModel().getWorldBound(), entry.getValue());
+                addDisplayable(damageDealer);
+            }
         }
         for (Path path : levelGenerator.getPathPieces(COMMANDER_LOCATION)) {
```

Comment:
Not sure why this in the initialization, shouldn't collision be checked in the main loop?